internal/service/metric/influxdb/influxdb.go

```diff
@@ -0,0 +1,116 @@
+package influxdb
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	_ "github.com/influxdata/influxdb1-client" // needed due to go mod bug
+	influxdbv2 "github.com/influxdata/influxdb1-client/v2"
+
+	"github.com/slok/grafterm/internal/model"
+	"github.com/slok/grafterm/internal/service/metric"
+)
+
+// ConfigGatherer is the configuration of the InfluxDB gatherer.
+type ConfigGatherer struct {
```

Comment:
What do you think about injecting the HTTP client and defaulting to Go's STD library when the client is not passed?

This way the client with the password, TLS, username... settings would be on the upper layer, we could inject any kind of configured client and the Influx gatherer doesn't need to know anything about clients except the address to connect to.

e.g.

https://github.com/slok/grafterm/blob/3b4bb0f580613a54023bab34f2672c4579ba7f9c/internal/service/metric/graphite/graphite.go#L19-L26

Reply:
Sounds great, i'll look at this. The first version was following the graphite example but i changed it to help with the tests. It now makes more sense why graphite had that.

Reply:
I couldn't find an easy way to do this (injecting the std http client). The library's implementation seems to be taking over the role: https://github.com/influxdata/influxdb1-client/blob/master/v2/client.go

Maybe i'm missing something?

Reply:
Sorry, you are right! I meant the influx default client, in this case, an influx client and not an STD client.

Reply:
Let me know if this is what you meant, just pushed an update.

Reply:
Yes, that's it! could you make a validation that the configuration data received in `ConfigGatherer` is ok? for example, if no addr error, if not client create a default one... whatever you think is better for each of the config paramaters.

Reply:
Default client added, i believe the influxdb client will already do the addr validation and return an appropriate error. Database validation also added so it's mandatory.